### ISSUE TITLE
where to host PNG files?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/vector-addition#119]
In the .js files for this sim, there are 10 comments that refer to PNG files that are hosted on user-images.githubusercontent.com.

For example, in Graph.js:

```js
  // See https://user-images.githubusercontent.com/42391580/61587030-0237a000-ab3f-11e9-9315-d1d2b4df6dd4.png
  // for an annotated drawing of the calculation.
```

... which looks like:

<img width="976" alt="screenshot_1347" src="https://user-images.githubusercontent.com/3046552/63464790-38598f80-c41d-11e9-8c94-4bb8a9d294c7.png">

Questions:
* Do these links expire?  
* Would it be preferable to put these PNG files in vector-addition/doc/ ?

**Comment by pixelzoom** [comment]
Here's the complete set of links.  All of these are intended to provide "a visual" for the purposes of code documentation.

https://user-images.githubusercontent.com/42391580/61587030-0237a000-ab3f-11e9-9315-d1d2b4df6dd4.png
https://user-images.githubusercontent.com/42391580/61587077-38295400-ab40-11e9-86e8-beeee7491ebf.png
https://user-images.githubusercontent.com/42391580/60908828-065be780-a23a-11e9-9a02-01a6ca52f73d.png
https://user-images.githubusercontent.com/42391580/61176588-88247b80-a580-11e9-96e4-b14e4d8e1212.png
https://user-images.githubusercontent.com/42391580/61176905-1e0ed500-a586-11e9-992b-d96757a6331b.png
 https://user-images.githubusercontent.com/42391580/60743952-f342d200-9f30-11e9-9a04-7b72ada15244.png
https://user-images.githubusercontent.com/42391580/61231836-27b14d80-a6ea-11e9-9238-e9562d6ab89d.png
https://user-images.githubusercontent.com/42391580/60760546-3619ae00-9ff4-11e9-8e91-508fc27f5e7c.png
https://user-images.githubusercontent.com/42391580/62811621-9bfbc880-babf-11e9-99b4-5fec8f25b84b.png
https://user-images.githubusercontent.com/42391580/62882976-4a865000-bcf0-11e9-9df4-cf1220efe314.png

**Comment by pixelzoom** [comment]
All of these URLs begin with `https://user-images.githubusercontent.com/42391580`.  Does `42391580` indicate a specific user account?

**Comment by ariel-phet** [comment]
in the /doc folder seems wise @pixelzoom

**Comment by pixelzoom** [comment]
I'll bring this up in dev meeting, to find out what others are doing, and to decide if we should standardize on one approach.

**Comment by samreid** [comment]
It's unclear whether those URLs are guaranteed to last long-term.  Also, I couldn't tell whether these images were added to user-images.githubusercontent.com by their presence in GitHub issue comments.  If the images are indeed in GitHub issue comments, it seems preferable to link to the GitHub issue comment (rather than the user-images.githubusercontent.com), since the issue URL seems like a long-term stable URL, and will also provide context and discussion around the image.

If there is no GitHub issue context around these images, then committing them to the sim repo/doc folder sounds best.

**Comment by brandonLi8** [comment]
@samreid I think the majority of these images were never in actual comments. I just solely made the images, pasted it onto an issue preview, and took advantage of github "user-content" hosting.

@pixelzoom said 
> Would it be preferable to put these PNG files in vector-addition/doc/ ?
I think it is better to have them as an actual url, so this would be fine with me they are still urls in the code. But at the same time, I think this only needs to be done if github "user-content" urls expire.

**Comment by pixelzoom** [comment]
Yes - If they live in vector-addition/doc/, they will still be referred to by URL in the code.

I've been unable to find any info on whether images that are not part of a GitHub issue will expire. I'm also concerned that if "42391580" refers to a specific user, then those images might be deleted if that user's GitHub account is ever deleted.  And all of this is difficult and time-consuming to test.  

So I think the safest approach would be to check such images into {{REPO}}/doc/, where they will be clearly owned and versionable by anyone associated with phetsims.

**Comment by samreid** [comment]
> I think it is better to have them as an actual url,

Can you please elaborate on why it is preferable to have them as a URL?

> so this would be fine with me they are still urls in the code

You mean they would point to localhost? or point to https://github.com/phetsims/vector-addition/blob/master/doc/?

> But at the same time, I think this only needs to be done if github "user-content" urls expire.

Entertaining the unlikely notion that github "user-content" might one day expire--if we are given no advance warning, then when they expire it will be too late to save the images, right?

> So I think the safest approach would be to check such images into {{REPO}}/doc/,

That sounds nice.

**Comment by samreid** [comment]
I also want to make sure @brandonLi8 is aware of PhET's precedent for using images in the other documentation.  For example, see https://github.com/phetsims/area-model-common/blob/master/doc/implementation-notes.md.  However, I don't mean to discourage you from commenting in the *.js files--just wanted to point out how we have matched documentation with images before.

**Comment by pixelzoom** [comment]
> Can you please elaborate on why it is preferable to have them as a URL?

In WebStorm for macOS, you can Command-B on a URL in a source file to open the URL in the browser. That's much easier than having to manually copy-paste or navigate. 

> You mean they would point to localhost? or point to https://github.com/phetsims/vector-addition/blob/master/doc/?

The latter, similar to a link to model.md, e.g. https://github.com/phetsims/vector-addition/blob/master/doc/model.md

**Comment by pixelzoom** [comment]
To expand on what @samreid said in ...

In the first item of the [Repository](https://github.com/phetsims/phet-info/blob/master/checklists/code_review_checklist.md#repository) section of the code-review checklist, it says:

```
 doc/
   images/
      *see annotation
 ...
*Any images used in model.md or implementation-notes.md should be added here. 
Images specific to aiding with documentation do not need their own license.
```

This should probably be extended to images referred to in source code.

**Comment by brandonLi8** [comment]
@samreid said:

>I also want to make sure @brandonLi8 is aware of PhET's precedent for using images in the other documentation. For example, see https://github.com/phetsims/area-model-common/blob/master/doc/implementation-notes.md. However, I don't mean to discourage you from commenting in the *.js files--just wanted to point out how we have matched documentation with images before.

I was not aware of that; thanks for letting me know.

**Comment by zepumph** [comment]
I think it may be easiest if these images ended up in the issue where the work was being done and committed to in the first place. Then you can link to that issue comment and it will exist forever, as well as giving more context (if desirable): the greater issue itself.

**Comment by pixelzoom** [comment]
8/22/19 dev meeting:
* Some of the images are out of date, they should be deleted.
* Move the ones that are useful to doc/ and link into code.
